File: src/monster.rs
#![allow(dead_code)]
#![allow(unused_variables)]
pub struct Monster {
    pub level: f64,
    pub resist: f64,
}

impl Monster {
    pub fn default() -> Self {
        Self {
            level: 93.0,
            resist: -0.1,
        }
    }

    pub fn new(level: f64, resist: f64) -> Self {
        Self {
            level,
            resist,
        }
    }
}

File: artifact-calculator/artifact-calculator-java/src/main/java/com/wenhao/calculator/monster/Monster.java
package com.wenhao.calculator.monster;

import com.wenhao.calculator.artifact.model.ArtifactSub;
import lombok.Data;

/**
 * Monster
 *
 * @author: Wenhao Tong
 * @date: 2023/6/19
 */
@Data
public class Monster {
    private Integer level;

    private  Double resist;

    public Monster() {
        level = 87;
        resist = 0.1;
    }

    public Monster(Integer level, Double resist) {
        this.level = level;
        this.resist = resist;
    }

    public void updateResist(ArtifactSub artifactSub) {
        if (artifactSub.getKeyword().isReduction()) {
            this.resist += artifactSub.getValue();
        }
    }
}
